frontend/ui/presets.lua

```diff
@@ -0,0 +1,325 @@
+--[[
+    This module provides a unified interface for managing presets across different KOReader modules.
+    It handles creation, loading, updating, and deletion of presets, as well as menu generation.
+
+    Usage:
+        local Presets = require("ui/presets")
+
+        -- 1. In your module's init() method, set up a preset object:
+            self.preset_obj = {
+                presets = G_reader_settings:readSetting("my_module_presets", {}),             -- or custom storage
+                cycle_index = G_reader_settings:readSetting("my_module_presets_cycle_index"), -- optional, only needed if cycling through presets
+                dispatcher_name = "load_my_module_preset",                                    -- must match dispatcher.lua entry
+                saveCycleIndex = function(this)                                               -- Save cycle index to persistent storage
+                    G_reader_settings:saveSetting("my_module_presets_cycle_index", this.cycle_index)
+                end,
+                buildPreset = function() return self:buildPreset() end,                       -- Closure to build a preset from current state
+                loadPreset = function(preset) self:loadPreset(preset) end,                    -- Closure to apply a preset to the module
+            }
+
+        -- 2. Implement required methods in your module:
+            function MyModule:buildPreset()
+                return {
+                    -- Return a table with the settings you want to save in the preset
+                    setting1 = self.setting1,
+                    setting2 = self.setting2,
+                    enabled_features = self.enabled_features,
+                }
+            end
+
+            function MyModule:loadPreset(preset)
+                -- Apply the preset settings to your module
+                self.setting1 = preset.setting1
+                self.setting2 = preset.setting2
+                self.enabled_features = preset.enabled_features
+                -- Update UI or perform other necessary changes
+                self:refresh()
+            end
+
+        -- 3. Create menu items for presets:
+            function MyModule:genPresetMenuItemTable(touchmenu_instance)
+                return Presets.genPresetMenuItemTable(
+                    self.preset_obj,                              -- preset object
+                    _("Create new preset from current settings"),    -- optional: custom text for UI menu
+                    function() return self:hasValidSettings() end,   -- optional: function to enable/disable creating presets
```

Comment:
(you might want to align the comments)
also probably: optional: function to enable/disable ~~creating presets~~ presets submenu

Reply:
(you might not ? :)